packages/hardhat-core/src/internal/core/providers/http.ts

```diff
@@ -37,7 +37,8 @@ export class HttpProvider extends EventEmitter implements EIP1193Provider {
     private readonly _url: string,
     private readonly _networkName: string,
     private readonly _extraHeaders: { [name: string]: string } = {},
-    private readonly _timeout = 20000
+    private readonly _timeout = 20000,
+    client: Pool | undefined = undefined
```

Comment:
What about 

```suggestion
    dispatcher: Dispatcher | undefined = undefined
```

and change the field

```
  private _client: Pool;
```

to

```
  private _dispatcher: Dispatcher;
```

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/2414/commits/c2893b9a6a701be53a294d726c52259a1887f761